minetest/minetest
# Startup screen code reorg

This PR is an attempt to reorganize start up screen code (main menu and loading screen).

I wanted to continue Kilbith's #9973 but I quickly got lost in code mess and duplication.

This PR gathers everything to manage startup screen except main menu in one single class. This class is will be responsible of drawing background (clouds or texture), header, footer and progress information (during loading).

Formerly, this was scattered in various files : guiEngine, RenderingEngine and Clientlauncher. Many other where using access to RenderingEngine singleton to update loading message. They all get now the instance of StartupScreen.

## To do

This PR is a Work in Progress

- There are still bugs to be fixed but mainly works (not tested all menu things).
- MenuTextureSource should be replaced by a StartupTextureSource;

In a (near or far) future: 
- Main menu could be reworked to offer possibility to plainly customize screen background and loading screen;
- Other scenes could be added instead of clouds;

## How to test

If you want, you can just compile, launch and observe main menu and loading screen. It should be close to what's already there.

Side effect (looks like an improvement): When launching a game, background is kept same as in main menu. So, game title and background are now displayed while loading.
